### ISSUE TITLE
Can't learn dodge skill thru dodging, no on dodge buffs from martial arts

### ISSUE THREAD

**Issue by 0novo** [CleverRaven/Cataclysm-DDA#66098]
### Describe the bug

The dodge skill cannot be trained and martial arts on dodge buffs will not trigger on a dodge

### Attach save file

n/a

### Steps to reproduce

1. Create a new character (and preferably mutate in Debug Invincibility, etc.)
2. Debug learn all martial arts, set unarmed combat skill to 10 (to enable martial arts techniques)
3. Spawn in a zombie, let said zombie attack you
4. Observe how your dodge skill does not increase when you dodge the zombie's attacks
5. Switch Judo, a martial art with an on dodge buff
6. Observe how you cannot get the "Perfect Position" on dodge buff
7. Switch Zui Quan, a martial art with an on dodge buff as well as a bonus to the amount of dodges one can do in a single turn
8. Observe how the dodge skill now increases and that now you can now get the on dodge buff "Drunken Dodging"

### Expected behavior

Dodge skill would be trained, martial arts buffs would trigger on dodge without needing to add bonus dodge attempts

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.2965 (22H2)
- Game Version: 41f05da [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]

### Additional context

I'm not the most code savvy, but I think this bug stems from the function Character::on_dodge in character.cpp, specifically: `if( !can_try_doge().success() ) { return; }`
I think Character::on_try_dodge() precedes Character::on_dodge() but has a consume_dodge_attempts() call, which means that the aforementioned conditional statement in Character::on_dodge() fails because can_try_doge() is called after consume_dodge_attempts()...?

**Comment by RenechCDDA** [comment]
>The dodge skill cannot be trained 

Confirmed.

>martial arts' on dodge buffs will not trigger on a dodge

Also confirmed.

**Comment by RGBDeadSilent** [comment]
Yep, having the same issue and I just started a dodge based character with martial arts. Commenting for visibility.

**Comment by Venera3** [comment]
Confirmed on `0.G-2023-06-06-0233` at the earliest, can't reproduce on `0.G-2023-06-06-0041` - meaning it's probably a consequence of something getting lost in #64224

**Comment by AzyWng** [comment]
I've tested the bug for myself - I'm on 2023-06-19-1330. Here's the screenshot of before...
![2023-06-19_22-26](https://github.com/CleverRaven/Cataclysm-DDA/assets/45751345/65e98692-a0ff-4e8c-95e2-7f3bc4d98f45)

And here's a screenshot after waiting for 30 minutes.
![2023-06-19_22-26_1](https://github.com/CleverRaven/Cataclysm-DDA/assets/45751345/c27324e7-d720-498b-b5f6-ba601a4d2f4d)

As you can see, my character's dodging skill hasn't raised a single percentage point from dodging boomer bile for half an hour. I don't know how my character earned 16% XP - perhaps it was from dodging a different kind of attack, like a standard melee.
